### PR TITLE
TIP-613: Fix as much integration tests as possible

### PR DESCRIPTION
**Pull request by damien-carcel**
**Description**

This PR intends to fix all failing integration tests that have no reason to. This means that some scenarios are still failing, but for a "good" reason.

- Some API tests are still failing, for different reasons:
    - some need the PQB to work for the result to be OK
    - some requires a product value counter we removed (it couldn't work anymore)
- Two completeness scenarios are failing because they use a behat catalog, which is currently not installable because of the PQB is broken
- A lot of PQB tests are failing, which is OK too as the PQB does not work at the moment
- `ProductValuesStorageIntegration::testProductWithAllAttributes` is failing because of metrics that have to much decimals, **and only because of that**
- `ProductSaverIntegration::testRawValuesForProductWithAllAttributes` is failing **only in PHP 7.0**

All these remaining API fails will be handled in another PR as this one is already complicated enough.

**Definition Of Done**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/Controller/ProductController.php [hunk 1]
```diff
@@ -272,9 +272,9 @@ public function createAction(Request $request)
     {
         $data = $this->getDecodedContent($request->getContent());
 
-        $data = $this->populateIdentifierProductValue($data);
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/Controller/ProductController.php:275]: normally you can drop the `populateIdentifierProductValue` method
we have done that in the ProductBuilder if I remember well

File: src/Pim/Bundle/ApiBundle/Controller/ProductController.php [hunk 2]
```diff
@@ -626,6 +626,7 @@ protected function checkAttributesParameters($attributes)
      */
     protected function populateIdentifierProductValue(array $data)
     {
+        // TODO : TIP-613; This does not work anymore, as ProductRepository::getIdentifierProperties() returns "identifier" instead of the attribute code.
```
Comment (jjanvier) [inline src/Pim/Bundle/ApiBundle/Controller/ProductController.php]: no problem, as method should be dropped

File: src/Pim/Bundle/ApiBundle/Controller/ProductController.php (hunk 1 above)
Comment (damien-carcel) [reply src/Pim/Bundle/ApiBundle/Controller/ProductController.php:275]: Nope, this is needed if we update the SKU of an existing product.

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Attribute/PartialUpdateAttributeIntergration.php [hunk 3]
```diff
@@ -6,15 +6,15 @@
 use Pim\Bundle\ApiBundle\tests\integration\ApiTestCase;
```
Comment (SamirBoulil) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Attribute/PartialUpdateAttributeIntergration.php:6]: I wonder how this test is Run `PartialUpdateAttributeInte*r*gration`

**Review by SamirBoulil — APPROVED** [review]
One small comment GTM

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Attribute/PartialUpdateAttributeIntergration.php (hunk 3 above)
Comment (damien-carcel) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Attribute/PartialUpdateAttributeIntergration.php:6]: Damn, I fix it badly…

**Review by anaelChardan — APPROVED** [review]

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Attribute/PartialUpdateAttributeIntergration.php (hunk 3 above)
Comment (damien-carcel) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Attribute/PartialUpdateAttributeIntergration.php:6]: Fixed

**Review by jjanvier — APPROVED** [review]

File: src/Pim/Bundle/ApiBundle/Controller/ProductController.php [hunk 4]
```diff
@@ -302,7 +302,7 @@ public function partialUpdateAction(Request $request, $code)
 
         if ($isCreation) {
             $this->validateCodeConsistency($code, $data);
-            $product = $this->productBuilder->createProduct();
+            $product = $this->productBuilder->createProduct($code);
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/Controller/ProductController.php:305]: as product identifier can be in the query path and/or in the body of the request, you should put ` $data['identifier'] = array_key_exists('identifier', $data) ? $data['identifier'] : $code;` before the condition `if ($isCreation)` and set `$product = $this->productBuilder->createProduct($data['identifier']);`

**Pull request merged**